drivers/shared/validators/validators.go

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package validators
+
+import (
+	"fmt"
+	"os/user"
+	"strconv"
+	"strings"
+)
+
+// IDRange defines a range of uids or gids (to eventually restrict)
+type IDRange struct {
+	Lower uint64 `codec:"from"`
+	Upper uint64 `codec:"to"`
+}
+
+// IDRangeValid is used to ensure that the configuration for ID ranges is valid.
+func IDRangeValid(rangeType string, deniedRanges string) error {
+	_, err := parseRanges(deniedRanges)
+	if err != nil {
+		return fmt.Errorf("invalid %s value %q: %v", rangeType, deniedRanges, err)
+	}
+
+	return nil
+}
+
+type userLookupFn func(string) (*user.User, error)
+
+// UserInRange is used when running a task to ensure the
+// given user is in the ID range defined in the task config
+func UserInRange(userLookupFn userLookupFn, usernameToLookup string, deniedHostUIDs, deniedHostGIDs string) error {
+
+	// look up user on host given username
+
+	u, err := userLookupFn(usernameToLookup)
+	if err != nil {
+		return fmt.Errorf("failed to identify user %q: %v", usernameToLookup, err)
+	}
+	uid, err := strconv.ParseUint(u.Uid, 10, 32)
+	if err != nil {
+		return fmt.Errorf("unable to convert userid %s to integer", u.Uid)
+	}
+
+	// check uids
+
+	uidRanges, err := parseRanges(deniedHostUIDs)
+	if err != nil {
+		return fmt.Errorf("invalid denied_host_uids value %q: %v", deniedHostUIDs, err)
```

Comment:
Wrap your errors wherever you can:

```suggestion
		return fmt.Errorf("invalid denied_host_uids value %q: %w", deniedHostUIDs, err)
```